### PR TITLE
consolidate submissions

### PR DESCRIPTION
**Pull request by snopoke**
Consolidate form submissions in code to use a single entry point and remove deprecated function.

@millerdev 
review by commit

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_domains.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 from django.test.utils import override_settings
 from casexml.apps.case.exceptions import IllegalCaseId
 from casexml.apps.case.models import CommCareCase
+from corehq.apps.receiverwrapper import submit_form_locally
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_domains.py]: F401 'submit_form_locally' imported but unused

File: corehq/form_processor/interfaces.py [hunk 2]
```diff
@@ -146,6 +145,7 @@ def get_case_xform_ids_from_couch(case_id):
 
     @staticmethod
     @to_generic
+    @unit_testing_only
     def post_xform(instance_xml, attachments=None, process=None, domain='test-domain'):
```
Comment (millerdev) [inline corehq/form_processor/interfaces.py]: Why put unit testing functions in a module with production code? Would it make sense to create a subclass of `FormProcessorInterface` with this method in `corehq.form_processor.tests`?

**Comment by millerdev** [comment]
Tests failed. Not sure if it's related to these changes or not:
```
======================================================================
FAIL: testMultipleCases (casexml.apps.phone.tests.test_sync_mode.SyncTokenUpdateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 204, in testMultipleCases
    self._testUpdate(sync_log.get_id, {"asdf": []})
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 100, in _testUpdate
    self.assertEqual(set(all_ids), sync_log.case_ids_on_phone)
AssertionError: set(['asdf']) != JsonSet([])
```

File: corehq/form_processor/interfaces.py (hunk 2 above)
Comment (snopoke) [inline corehq/form_processor/interfaces.py]: Completely agree, that is the plan down the line

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by czue** [comment]
:dancer:

**Pull request merged**